contentcuration/contentcuration/views/files.py

```diff
@@ -236,6 +241,80 @@ def exercise_image_upload(request):
     }))
 
 
+def subtitle_upload(request):
+    # File will be converted to VTT format
+    ext = file_formats.VTT
+    language_id = request.META.get('HTTP_LANGUAGE')
+    content_file = request.FILES.values()[0]
+
+    with NamedTemporaryFile() as temp_file:
+        try:
+            converter = build_subtitle_converter(unicode(content_file.read(), 'utf-8'))
+            convert_language_code = language_id
+
+            # We're making the assumption here that language the user selected is truly the caption
+            # file's language if it's unknown
+            if len(converter.get_language_codes()) == 1 \
+                    and converter.has_language(LANGUAGE_CODE_UNKNOWN):
+                converter.replace_unknown_language(language_id)
+
+            # determine if the request language exists by another code, otherwise we can't continue
+            if not converter.has_language(convert_language_code):
+                for language_code in converter.get_language_codes():
+                    language = getlang_by_alpha2(language_code)
+                    if language and language.code == language_id:
+                        convert_language_code = language_code
+                        break
+                else:
+                    return HttpResponseBadRequest(
+                        "Language '{}' not present in subtitle file".format(language_id))
+
+            converter.write(temp_file.name, convert_language_code)
+        except InvalidSubtitleFormatError as ex:
+            return HttpResponseBadRequest("Subtitle conversion failed: {}".format(ex))
+
+        temp_file.seek(0)
+        converted_file = DjFile(temp_file)
+
+        checksum = get_hash(converted_file)
+        size = converted_file.size
+        request.user.check_space(size, checksum)
+
+        file_object = File(
+            file_size=size,
+            file_on_disk=converted_file,
+            checksum=checksum,
+            file_format_id=ext,
+            original_filename=request.FILES.values()[0]._name,
+            preset_id=request.META.get('HTTP_PRESET'),
+            language_id=language_id,
+            uploaded_by=request.user,
+        )
+        file_object.save()
+
+    return HttpResponse(json.dumps({
+        "success": True,
+        "filename": str(file_object),
+        "file": JSONRenderer().render(FileSerializer(file_object).data)
+    }))
+
+
+@authentication_classes((TokenAuthentication, SessionAuthentication))
+@permission_classes((IsAuthenticated,))
+def multilanguage_file_upload(request):
```

Comment:
Thanks for updating! One more thing to make this more flexible, could you move the [file creation logic](https://github.com/learningequality/studio/pull/1500/files#diff-ce4f9c1624caa8107421afda35e090e0R283) here as well and make the subtitle conversion code return a file? That way any future non-subtitle, multi-language files will still get created

Reply:
I can do that, but I do have some thoughts. Do you think it's worthwhile to build out some handling for a case that doesn't exist? In my eyes, that would mean introducing some unreachable code. Also, if that default implementation doesn't work for the next file format, I think it would be easier to identify if it was unimplemented. What are your thoughts?

Reply:
Sorry for the late response! That's a good point- I guess we could build out the cases as we go. Although, will vtt files still go through the conversion process?

Reply:
All subtitle files will go through the converter as it will verify the type (that VTT files are VTT) and it standardizes the VTT output.